client/gutenberg/extensions/presets/jetpack/editor-shared/get-site-fragment.js

```diff
@@ -26,5 +26,14 @@ export default function getSiteFragment() {
 		return siteFragment || null;
 	}
 
+	// Gutenberg in Jetpack adds a site fragment in the initial state
+	if (
+		window &&
+		window.Jetpack_Editor_Initial_State &&
+		window.Jetpack_Editor_Initial_State.siteFragment
+	) {
+		return window.Jetpack_Editor_Initial_State.siteFragment;
+	}
+
```

Comment:
Hmm, we already have a rather similar check a few lines above:

https://github.com/Automattic/wp-calypso/blob/fbec9c9553ce0220e8bca49a6c29ae6cebb8e17e/client/gutenberg/extensions/presets/jetpack/editor-shared/get-site-fragment.js#L17-L19

Is `window._currentSiteId` not available in Jetpack? Can we converge things so we only one source of truth for wp-admin (both JP and WP.com) at least? I.e. either `window._currentSiteId` _or_ `window.Jetpack_Editor_Initial_State.siteFragment` in both cases?

Reply:
fwiw - `window._currentSiteId` is `undefined` in JN's wp-admin block-editor.

Reply:
TBH, I'd really prefer not having to rely on `window._currentSiteId` at all, and have this proposed solution work for wp-admin both in Jetpack and WP.com.

Reply:
This is a good discussion. 

If I may, I have another point I'd like to bring :)

The current solution (`Jetpack_Editor_Initial_State.siteFragment`) does not actually work for WordPress.com right now, it will need to be adapted. See https://github.com/Automattic/jetpack/pull/11347 for more details).

That said, using the site ID seems easier to me, the patch I linked to above would end up simpler. That would however require that we update this PR as well to not use `window.Jetpack_Editor_Initial_State.siteFragment`, but instead use `window.Jetpack_Editor_Initial_State.jetpack.blog_id`.

Reply:
@ockham this is not the site id. Yeah I can refactor it to use the site id. The reasons why I used the site fragment instead are because I didn't want to possibly conflict with something intended for WPCOM only (since _currentSiteId doesn't exist in Jetpack for some odd reason) and looks more understandable in the URL.

Reply:
Just to avoid confusion: there's a comment above which says:

> // WP.com wp-admin exposes the site ID in window._currentSiteId

And that's completely intentional - that parameter has never intended to be present in Jetpack wp-admin; we actually used that, knowing it's only for WP.com wp-admins.